motionSetPointsApp/src/tests/testSetPoint.cc

```diff
@@ -0,0 +1,451 @@
+    }
+
+    TEST(setPoint, GIVEN_file_with_single_row_and_single_coord_WHEN_loadFile_called_THEN_single_row_and_coord_with_correct_data) {
+        auto testFilename = "my_file";
+        auto envFilename = "TEST";
+        std::vector<std::string> fileLines = { "Position    2" };
+        MockFileIO mockFile;
+        
+        createMockFile(&mockFile, testFilename, fileLines);
+
+        loadFile(&mockFile, testFilename, envFilename, 1);
+
+        auto table = getTable(envFilename);
+
+        ASSERT_EQ(table.rows.size(), 1);
+        ASSERT_THAT(table.rows[0].coordinates, ElementsAre(2));
+    }
+
+    TEST(setPoint, GIVEN_file_with_single_row_and_three_coords_WHEN_loadFile_called_THEN_single_row_and_coord_with_correct_data) {
+        auto testFilename = "my_file";
+        auto envFilename = "TEST";
+        std::vector<std::string> fileLines = { "Position    2    9   8" };
+        MockFileIO mockFile;
+        
+        createMockFile(&mockFile, testFilename, fileLines);
+
+        loadFile(&mockFile, testFilename, envFilename, 3);
+
+        auto table = getTable(envFilename);
+
+        ASSERT_EQ(table.rows.size(), 1);
+        ASSERT_THAT(table.rows[0].coordinates, ElementsAre(2, 9, 8));
+    }
+
+    TEST(setPoint, GIVEN_file_with_multiple_rows_and_single_coord_WHEN_loadFile_called_THEN_multiple_rows_and_coord_with_correct_data) {
+        auto testFilename = "my_file";
+        auto envFilename = "TEST";
+        std::vector<std::string> fileLines = { "Position    2", "Position_2    7" };
+        MockFileIO mockFile;
+
+        createMockFile(&mockFile, testFilename, fileLines);
+
+        loadFile(&mockFile, testFilename, envFilename, 1);
+
+        auto table = getTable(envFilename);
+
+        ASSERT_EQ(table.rows.size(), 2);
+        ASSERT_THAT(table.rows[0].coordinates, ElementsAre(2));
+        ASSERT_THAT(table.rows[1].coordinates, ElementsAre(7));
+    }
+
+    TEST(setPoint, GIVEN_file_with_inconsistent_coord_WHEN_loadFile_called_THEN_table_cleared) {
+        auto testFilename = "my_file";
+        auto envFilename = "TEST";
+        std::vector<std::string> fileLines = { "Position    2", "Position_2    7    9" };
+        MockFileIO mockFile;
+
+        createMockFile(&mockFile, testFilename, fileLines, false);
+
+        loadFile(&mockFile, testFilename, envFilename, 1);
+
+        auto table = getTable(envFilename);
+
+        ASSERT_EQ(table.rows.size(), 0);
+    }
+
+    TEST(setPoint, GIVEN_file_with_multiple_positions_with_the_same_name_WHEN_loadFile_called_THEN_table_cleared) {
+        auto testFilename = "my_file";
+        auto envFilename = "TEST";
+        std::vector<std::string> fileLines = { "Position    2", "Position    7" };
+        MockFileIO mockFile;
+
+        createMockFile(&mockFile, testFilename, fileLines, false);
+
+        loadFile(&mockFile, testFilename, envFilename, 1);
+
+        auto table = getTable(envFilename);
+
+        ASSERT_EQ(table.rows.size(), 0);
```

Comment:
If the switch is made to producing an error can we test the error condition in here